gap/ideals/ideals.gd

```diff
@@ -8,7 +8,20 @@
 #############################################################################
 
 DeclareSynonymAttr("GeneratorsOfSemigroupIdeal", GeneratorsOfMagmaIdeal);
+
 DeclareGlobalFunction("SemigroupIdeal");
+DeclareGlobalFunction("LeftSemigroupIdeal");
+DeclareGlobalFunction("RightSemigroupIdeal");
+DeclareGlobalFunction("AnySemigroupIdealInputParsing");
```

Comment:
```suggestion
DeclareGlobalFunction("AnySemigroupIdealInputParsing");
```
I'd suggest removing this from here, and instead doing 

~~~
BindGlobal("_AnySemigroupIdealInputParsing", ...)
~~~

in the `.gi` file instead. Note also the `_` in the name (either do this or just put the function into the record `SEMIGROUPS`, by doing:

~~~
SEMIGROUPS.AnySemigroupIdealInputParsing := function(...
~~~
again in the `.gi` file. Same goes for all of the "internal" functions that start with "Any"